File: client/src/app/models/admin-message.ts
import { AbstractModel } from './abstract-model';

/**
 * Admin-authored broadcast message delivered as an inbox notification.
 */
export class AdminMessage extends AbstractModel {
  title: string;
  text: string;

  public static deserialize(payload: any): AdminMessage {
    const message = new AdminMessage();
    AbstractModel.deserializeAbstractAttributes(message, payload);
    message.title = payload.title;
    message.text = payload.text;
    return message;
  }

  public static serialize(message: AdminMessage): any {
    return {
      title: message.title,
      text: message.text,
    };
  }
}

File: server/src/models/admin_message.py
from extensions import db
from models.base_entity import BaseEntity


class AdminMessage(BaseEntity):
    """
    Admin-authored broadcast message shown as an inbox notification with a detail dialog.
    """

    __tablename__ = "admin_messages"

    title = db.Column(db.String(200), nullable=False)
    text = db.Column(db.Text(), nullable=False)
